src/Core/Grid/Definition/Factory/CmsPageDefinitionFactory.php

```diff
@@ -0,0 +1,330 @@
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class responsible for providing columns, filters, actions for cms page list.
+ */
+class CmsPageDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    const GRID_ID = 'cms_page';
+
+    /**
+     * @var int
+     */
+    private $cmsCategoryParentId;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher,
+        CommandBusInterface $queryBus,
+        RequestStack $requestStack
+    ) {
+        parent::__construct($hookDispatcher);
+
+        $this->queryBus = $queryBus;
+        $this->requestStack = $requestStack;
+
+        $this->setCmsPageCategoryParentId($requestStack);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        $cmsCategoryName = $this->queryBus->handle(new GetCmsPageCategoryNameForListing());
+
+        return $this->trans(
+            'Pages in category "%name%"',
+            array('%name%' => $cmsCategoryName),
+            'Admin.Design.Feature'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        return (new ColumnCollection())
+            ->add((new BulkActionColumn('bulk'))
+                ->setOptions([
+                    'bulk_field' => 'id_cms',
+                ])
+            )
+            ->add((new DataColumn('id_cms'))
+                ->setName($this->trans('ID', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'id_cms',
+                ])
+            )
+            ->add((new DataColumn('link_rewrite'))
+                ->setName($this->trans('URL', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'link_rewrite',
+                ])
+            )
+            ->add((new DataColumn('meta_title'))
+                ->setName($this->trans('Title', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'meta_title',
+                ])
+            )
+            ->add((new DataColumn('head_seo_title'))
+                ->setName($this->trans('Meta title', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'head_seo_title',
+                ])
+            )
+            ->add((new DataColumn('position'))
```

Comment:
No `PositionColumn` ? no need to order this list maybe?

Reply:
I postponed the usage of `PositionColumn`. I referred few problems here https://github.com/PrestaShop/PrestaShop/pull/12974

Reply:
Alright, I answered some of your questions in https://github.com/PrestaShop/PrestaShop/pull/12974#issuecomment-476614733